storage/remote/client.go

```diff
@@ -38,30 +39,54 @@ const maxErrMsgLen = 256
 
 var userAgent = fmt.Sprintf("Prometheus/%s", version.Version)
 
-// Client allows reading and writing from/to a remote HTTP endpoint.
-type Client struct {
+// client allows reading and writing from/to a remote HTTP endpoint.
+type client struct {
 	remoteName string // Used to differentiate clients in metrics.
 	url        *config_util.URL
 	client     *http.Client
 	timeout    time.Duration
+
+	readQueries prometheus.Gauge
 }
 
-// ClientConfig configures a Client.
+// ClientConfig configures a client.
 type ClientConfig struct {
 	URL              *config_util.URL
 	Timeout          model.Duration
 	HTTPClientConfig config_util.HTTPClientConfig
 }
 
-// NewClient creates a new Client.
-func NewClient(remoteName string, conf *ClientConfig) (*Client, error) {
+// TODO(bwplotka): Add streamed chunked remote read (add issue link).
```

Comment:
did you already open this issue?

Reply:
yes, will update link: https://github.com/prometheus/prometheus/issues/5926 forgot